packages/vite/src/node/plugins/resolve.ts

```diff
@@ -816,7 +816,7 @@ export function tryNodeResolve(
     exclude?.includes(pkgId) ||
     exclude?.includes(nestedPath) ||
     SPECIAL_QUERY_RE.test(resolved) ||
-    (!isBuild && ssr) ||
+    // (!isBuild && ssr) ||
```

Comment:
If this condition exist, SSR dep optimizer never runs in dev.
I guess it should run according to #8917.

(I'll delete this line if it's correct)

Reply:
This does seem like we should remove it too, but @patak-dev might know more about this.

Reply:
We need this line, could you add a comment there:
```
// During dev SSR, we don't have a way to reload the module graph if 
// a non-optimized dep is found. So we need to skip optimization here.
// The only optimized deps are the ones explicitly listed in the config.
```